themes/default/content/docs/guides/testing/integration.md

```diff
@@ -154,3 +151,22 @@ unlock new automation capabilities that improve your team's productivity and con
 A minimal example of using Pulumi's Go integration test framework is available in the examples repository: [Integration Testing in Go](https://github.com/pulumi/examples/tree/05ae8e1803d7f44cecac69589175e416e421cdfe/testing-integration).
 
 Our own [integration test suite](https://github.com/pulumi/examples/blob/05ae8e1803d7f44cecac69589175e416e421cdfe/misc/test/examples_test.go) can be a source of further inspiration.
+
+## Using Automation API for Integration Tests
+
+Pulumi's Automation API provides another option for writing end-to-end integration tests. It is not a purpose-built integration testing framework as with above, but you can still accomplish the same objectives of basic correctness testing, resource validation, and runtime testing. Automation API is available in the Node, Python, .NET, and Go SDKs.
+
+At a high level, you will want to create a stack, setup stack config, deploy the stack, perform any resource validation or runtime checks, and finally teardown the stack. The [`localProgram-tsnode-mochatests` example](https://github.com/pulumi/automation-api-examples/tree/main/nodejs/localProgram-tsnode-mochatests) provides one example for how to setup a stack, perform runtime validation checks, and then tear down the stack as part of a test.
+
+In order to perform resource validation, you can export the stack and examine the resulting resources. For example:
+
+**automation.ts**
+```typescript
+export async function getDeployment(): Promise<Deployment> {
+  const stack = await LocalWorkspace.createOrSelectStack(args);
+  
+  return stack.exportStack();
+}
+```
+
+You can then iterate through the `Deployment` object to check that the expected resources and property values are present.
```

Comment:
Suggested addition: 

We even write tests inside pulumi using Automation API and language-specific frameworks like `mocha`. Take a look at the source for yourself:

- [NodeJS tests](https://github.com/pulumi/pulumi/blob/d93e5acf675fe8f63cfcf0c028cf996f97baf621/sdk/nodejs/tests/automation/localWorkspace.spec.ts#L189)
- [Go tests](https://github.com/pulumi/pulumi/blob/d93e5acf675fe8f63cfcf0c028cf996f97baf621/sdk/go/x/auto/local_workspace_test.go#L734)
- [Python tests](https://github.com/pulumi/pulumi/blob/0c4e1a33e058a5b129ba009c5a58793200c09fff/sdk/python/lib/test/automation/test_local_workspace.py#L291)
- [C# tests](https://github.com/pulumi/pulumi/blob/0e2d9185870d1cc343c3f75499ec4265ba072292/sdk/dotnet/Pulumi.Automation.Tests/LocalWorkspaceTests.cs#L345)

(URLs will change with 3.0)

Reply:
Great suggestion!